lightningd/peer_control.c

```diff
@@ -875,6 +875,55 @@ void activate_peers(struct lightningd *ld)
 		activate_peer(p);
 }
 
+/* Peer has been released from gossip. */
+static void gossip_peer_disconnected (struct subd *gossip,
+				 const u8 *resp,
+				 const int *fds,
+				 struct command *cmd) {
+	if (!fromwire_gossipctl_peer_disconnect_reply(resp)) {
+		if (!fromwire_gossipctl_peer_disconnect_replyfail(resp))
+			fatal("Gossip daemon gave invalid reply %s",
+			      tal_hex(gossip, resp));
+		command_fail(cmd, "Peer not connected");
```

Comment:
There are actually 2 possible errors here, either peer is not connected, or peer has a channel and is doing something other than gossiping.  We can add a `bool` to `gossipctl_peer_disconnect_replyfail` to differentiate.